pkg/azure/client.go

```diff
@@ -100,6 +101,10 @@ func (az *azClient) UpdateGateway(appGwObj *n.ApplicationGateway) (err error) {
 }
 
 func (az *azClient) GetPublicIP(resourceID string) (n.PublicIPAddress, error) {
+	if ip, ok := az.memoizedIPs[resourceID]; ok {
```

Comment:
we need set `memoizedIPs`

Reply:
Good point.  I'm bringing small chunks of a larger PR and missed that. Thanks.